### ISSUE TITLE
Editor functions

### ISSUE THREAD

**Issue by eriksvedang** [carp-lang/Carp#1028]
info-at-point
type-at-point
goto-definition

**Comment by jacereda** [comment]
What about using LSP

**Comment by tealeg** [comment]
I have an LSP implementation in the back of my mind as a possible project, but it's certainly much bigger than what's implied here.

**Comment by jacereda** [comment]
Looks like starting wouldn't be too hard:

https://github.com/alanz/lsp/blob/master/example/Simple.hs

**Comment by eriksvedang** [comment]
Sure, that sounds like an even better idea!

**Comment by eriksvedang** [comment]
Can we use that for flycheck (currently supported via --check) too, then?

**Comment by jacereda** [comment]
I think so. AFAIK flycheck integrates with LSP.

**Comment by tealeg** [comment]
@eriksvedang LSP is well supported in Emacs at this point - I use it for Go, Python and Rust very happily.  Flycheck works perfectly.

**Comment by eriksvedang** [comment]
Awesome, I'm VERY excited about this approach then!

**Comment by tealeg** [comment]
Re @jacereda 's linked Haskel code - as a general question, but wouldn't it be good to implement this in Carp rather than Haskel?  The more tooling is in a language other than Carp, the bigger barrier there is to people working on it (in many cases, it requires learning two languages instead of one).

**Comment by hellerve** [comment]
That is true, but considering that the Carp compiler is written in Haskell, hooking into it (which might be required) will probably be simpler from within Haskell.

**Comment by tealeg** [comment]
@hellerve I guess that leads to a bigger topic - a self-hosted compiler, but that's a whole different ball game ;-)

**Comment by jacereda** [comment]
AFAIK LSP is quite messy, piggybacking into whatever that project has to offer can be a great timesaver.

**Comment by jgarte** [comment]
> @eriksvedang LSP is well supported in Emacs at this point - I use it for Go, Python and Rust very happily. Flycheck works perfectly.

emacs 29 now has lsp built in

**Comment by eriksvedang** [comment]
That is great news!
